apollo-compiler/src/test/kotlin/com/apollographql/apollo3/compiler/CodegenTest.kt

```diff
@@ -181,6 +216,18 @@ class CodegenTest(private val folder: File) {
              */
             file.isDirectory && (filterRegex == null || filterRegex.matchEntire(file.name) != null)
           }
+          .flatMap { file ->
+            if (fragmentsCodegenMode == null) {
+              listOf(
+                  arrayOf(file, FragmentsCodegenMode.FragmentsAsInterfaces),
+                  arrayOf(file, FragmentsCodegenMode.FragmentsAsDataClasses)
```

Comment:
Thinking more about this, it is a bit wasteful to run all the tests, event for operations without fragments like custom scalars and others test. On the other hand, it's still useful to be able to compare the generated code for operations where there are fragments, I'll try to think about something.

Reply:
I put a fix to run 2 tests for queries with fragments only.